### PR TITLE
Modifying keypoint_rcnn.py for keypoint_predictor issue

### LINKED ISSUE #5126: Bug in keypoint_rcnn initializer
### 🐛 Describe the bug

Line 1890 in keypoint_rcnn
"
        if num_classes is not None:
            if keypoint_predictor is not None:
                raise ValueError("num_classes should be None when keypoint_predictor is specified")
"
Shouldn't we change num_classes to num_keypoints in this case?

### Versions

Newest

cc @datumbox

### PR DESCRIPTION
**Pull request by themendu**
Fixes #5126

Hi @datumbox, I have created a PR for the issue [#5126](https://github.com/pytorch/vision/issues/5126). This is my first PR. Please review and comment on it. Thank you!

cc @datumbox

### CODE REVIEW

File: torchvision/models/detection/keypoint_rcnn.py [hunk 1]
```diff
@@ -188,16 +188,19 @@ def __init__(
...
         if min_size is None:
             min_size = (640, 672, 704, 736, 768, 800)
 
-        if num_classes is not None:
+        if num_keypoints is not None:
             if keypoint_predictor is not None:
-                raise ValueError("num_classes should be None when keypoint_predictor is specified")
+                raise ValueError("num_keypoints should be None when keypoint_predictor is specified")
+
+        if num_keypoints is None:
```
Comment (datumbox) [inline torchvision/models/detection/keypoint_rcnn.py:202]: Nit: Simplify by
```suggestion
        else:
```

**Review by datumbox — COMMENTED** [review]
@themendu Thanks for the PR, looks good overall. Just one nitpick, let me know what you think.

**Review by themendu — COMMENTED** [review]
I agree with @datumbox for directly including the else statement without complicating the issue.

**Review by datumbox — APPROVED** [review]
LGTM @themendu. Let's wait for green CI and I'll merge.

Thanks for your first contribution, looking forward to more on the future. :)

**Pull request merged**